apps/nextjs-website/react-components/components/Form/Form.tsx

```diff
@@ -107,80 +131,144 @@ const Form = (props: FormProps & { onSubmit: (data: FormData) => void }) => {
-              sx={{ backgroundColor: 'white', color: 'black' }}
-            />
-          </FormControl>
-        </Grid>
+        {showFirstName && (
+          <Grid item xs={12} sm={6}>
+            <FormControl fullWidth>
+              <OutlinedInput
+                placeholder='Nome'
+                name='firstName'
+                value={formData.firstName}
+                onChange={handleInputChange}
+                sx={{ backgroundColor: 'white', color: 'black' }}
+              />
+            </FormControl>
+          </Grid>
+        )}
+        {showLastName && (
+          <Grid item xs={12} sm={6}>
+            <FormControl fullWidth>
+              <OutlinedInput
+                placeholder='Cognome'
+                name='lastName'
+                value={formData.lastName}
+                onChange={handleInputChange}
+                sx={{ backgroundColor: 'white', color: 'black' }}
+              />
+            </FormControl>
+          </Grid>
+        )}
+        {showEmail && (
+          <Grid item xs={12}>
+            <FormControl fullWidth>
+              <OutlinedInput
+                placeholder='Indirizzo e-mail'
+                name='email'
+                value={formData.email}
+                onChange={handleInputChange}
+                sx={{ backgroundColor: 'white', color: 'black' }}
+              />
+            </FormControl>
+          </Grid>
+        )}
+        {showOrganization && (
+          <Grid item xs={12}>
+            <FormControl fullWidth>
+              <OutlinedInput
+                placeholder='Nome ente'
+                name='organization'
+                value={formData.organization}
+                onChange={handleInputChange}
+                sx={{ backgroundColor: 'white', color: 'black' }}
+              />
+            </FormControl>
+          </Grid>
+        )}
       </Grid>
-      <Typography
-        variant='body2'
-        sx={{ mb: 2, position: 'relative', zIndex: 3, color: textColor }}
-      >
-        {privacyText}{' '}
-        <Link
-          href={privacyLink}
-          sx={{ color: linkColor, textDecorationColor: linkColor }}
+      {checkboxTitle && (
+        <Typography
+          variant='h6'
+          gutterBottom
+          sx={{
+            position: 'relative',
+            zIndex: 3,
+            mb: 2,
+            mt: 4,
+            color: textColor,
+            fontWeight: '700',
+          }}
         >
-          {privacyLinkText}
-        </Link>
-      </Typography>
-      <Divider
-        sx={{
-          mb: 2,
-          position: 'relative',
-          zIndex: 3,
-          backgroundColor: textColor,
-        }}
+          {checkboxTitle}
+        </Typography>
+      )}
+      <FormCategories
+        formCategories={updatedFormCategories}
+        textColor={textColor}
+        borderColor={borderColor}
+        selectedOption={selectedOption}
+        handleRadioChange={handleRadioChange}
       />
```

Comment:
```suggestion
{updatedFormCategories.length > 0 && (
      <FormCategories
        formCategories={updatedFormCategories}
        textColor={textColor}
        borderColor={borderColor}
        selectedOption={selectedOption}
        handleRadioChange={handleRadioChange}
      />
)}
```
I would add a check in case no formCategories are passed.